apps/paymaster-proxy/src/adminApi/createAdminRouter.ts

```diff
@@ -0,0 +1,186 @@
+          entityId,
+          state: 'enabled',
+        })
+      newApiKey = createApiKeyResponse.apiKey
+    } catch (e) {
+      logger.error(e, 'Failed to create API key')
+      return res.status(500).json({
+        error: 'Failed to create API key',
+      })
+    }
+
+    // create new policy on Alchemy
+    let newPaymasterProviderPolicy: AlchemyGasManagerPolicy
+    try {
+      newPaymasterProviderPolicy =
+        await alchemyGasManagerAdminClient.createPolicy({
+          policyName: `paymaster-proxy:${envVars.DEPLOYMENT_ENV}:${chainId}:${newApiKey.id}`,
+          policyType: 'sponsorship',
+          rules: getAlchemyGasManagerDefaultRules(),
+        })
+    } catch (e) {
+      logger.error(e, 'Failed to create Alchemy Gas Manager policy')
+      return res.status(500).json({
+        error: 'Failed to create Alchemy Gas Manager policy',
+      })
+    }
+
+    // save policy ID and API key in DB
+    try {
+      await createSponsorshipPolicy(database, {
+        apiKeyId: newApiKey.id,
+        chainId: chainId.toString(),
+        providerType: 'alchemy',
+        providerMetadata: {
+          policyId: newPaymasterProviderPolicy.policyId,
+        },
+      })
+    } catch (e) {
+      logger.error(e, 'Failed to save policy ID and API key in DB')
+      return res.status(500).json({
+        error: 'Failed to save policy ID and API key in DB',
+      })
+    }
+
+    return res.json({
+      apiKey: newApiKey.key,
+    })
+  })
+
+  router.post('/deletePaymasterApiKey', async (req, res) => {
+    const paramsParseResult = z
+      .object({
+        chainId: z
+          .number()
+          .int()
+          .refine((chainId) => !!chainConfigByChainId[chainId], {
+            message: 'Unsupported chainId',
+          }),
+        apiKey: z.string(),
+      })
+      .safeParse(req.body)
+
+    if (!paramsParseResult.success) {
+      return res.status(400).json({
+        error: 'Invalid params',
+      })
+    }
+
+    const alchemyGasManagerAdminClient =
+      chainConfigByChainId[paramsParseResult.data.chainId]
+        .alchemyGasManagerAdminClient
+
+    const { apiKey } = paramsParseResult.data
+
+    const verifyApiKeyResult =
+      await apiKeyServiceClient.keys.verifyApiKey.query({
+        key: apiKey,
+      })
+
+    if (!verifyApiKeyResult || !verifyApiKeyResult.apiKey) {
+      return res.status(400).json({
+        error: 'Invalid API key',
+      })
+    }
+    const apiKeyId = verifyApiKeyResult.apiKey.id
+
+    const sponsorshipPolicy = await getSponsorshipPolicyForApiKeyId(
+      database,
+      apiKeyId,
+    )
+
+    if (!sponsorshipPolicy) {
+      return res.status(400).json({
+        error: 'No policy found for API key',
+      })
+    }
+
+    await alchemyGasManagerAdminClient.deletePolicy(
+      sponsorshipPolicy.providerMetadata.policyId,
+    )
+
+    await apiKeyServiceClient.keys.deleteApiKey.mutate({
+      id: sponsorshipPolicy.apiKeyId,
+    })
+
```

Comment:
should there be an rpc to update the sponsorship policy in the paymaster db?

Reply:
don't think we need one just yet but can add once we start needing it!